runtime/src/lib.rs

```diff
@@ -743,6 +775,45 @@ mod tests {
         });
     }
 
+    fn test_metrics(runner: impl Runner, context: impl Spawner + Metrics) {
+        runner.start(async move {
+            // Assert label
+            assert_eq!(context.label(), "");
+
+            // Register a metric
+            let counter = Counter::<u64>::default();
+            context.register("test", "test", counter.clone());
+
+            // Increment the counter
+            counter.inc();
+
+            // Encode metrics
+            let buffer = context.encode();
+            println!("{}", buffer);
+            assert!(buffer.contains("test_total 1"));
+
+            // Nested context
+            let context = context.with_label("nested");
+            let nested_counter = Counter::<u64>::default();
+            context.register("test", "test", nested_counter.clone());
+
+            // Increment the counter
+            nested_counter.inc();
+
+            // Encode metrics
+            let buffer = context.encode();
+            assert!(buffer.contains("nested_test_total 1"));
+            assert!(buffer.contains("test_total 1"));
+        });
+    }
+
+    fn test_metrics_label(runner: impl Runner, context: impl Spawner + Metrics) {
+        runner.start(async move {
+            context.with_label(METRICS_PREFIX);
+            panic!("should not be possible to create a context with the METRICS_PREFIX");
```

Comment:
You don't need to "absorb" the panic from the previous statement somehow for this test to pass?

Reply:
Correct. This is a mistake.